src/lib.rs

```diff
@@ -1098,6 +1098,21 @@ mod test {
     }
 
     #[test]
+    #[should_panic(expected = "Unexpected value")]
```

Comment:
My earlier comment was pointing out that panicking (due to `unwrap()`) is not the behaviour we want, so this test should not be expecting it.